### ISSUE TITLE
Finite loop detector misfires when used with mouseDragged

### ISSUE THREAD

**Issue by lgto4** [Khan/live-editor#518]
The KA finite loop detector misfires when a CPU intensive function is invoked from mouseDragged.  Invoking the same function from draw is fine (as if anything associated with the loop detector is "fine"). 

The thrown error is not recognised by Yoko Oh Noes.

A simple test program is at https://www.khanacademy.org/computer-programming/wat4/5728076905971712    This is using Chrome on LinuxMint (Ubuntu).

**Comment by kevinbarabash** [comment]
I wasn't able to repro this probably b/c my computer is too fast.  What does "the thrown error is not recognised" mean?

**Comment by lgto4** [comment]
The complaint from Yoko Oh Noes is:
 "Oh noes!
Show me where"
No mention of a problem.

The number one problem with the KA loop detector is that its behavior is a function of the host machine's speed.

**Comment by brew4321** [comment]
![error](https://cloud.githubusercontent.com/assets/15134841/11764808/b065f4ee-a0fb-11e5-85bc-d2eeac2baf30.jpg)

**Comment by brew4321** [comment]
While trying to reproduce it, I put in a variable `s` to scale that outer loop by an arbitrary amount, making compute arbitrarily more complex... then slowly increased it while messing around. Also after reproducing it I was able to cause the error to revert to the normal "A for loop is taking too long to run. Perhaps you have a mistake in your code?" message. (I think just by editing the code..)
I put in both of these:
```draw = doit;
//mouseDragged = doit;  /* WAT?  This works fine when assigned to draw */```
and was toggling which one was commented when it reverted.

**Comment by brew4321** [comment]
Another program is displaying a blank error: https://www.khanacademy.org/computer-programming/gingerbread-maker/4650907356233728

**Comment by Potato42** [comment]
Came here to report blank error.  My guess is the blank error is due to https://github.com/Khan/live-editor/commit/57130931aa8ded361a081b7cfc803c3bf4d317e0, but the only place I've found so far that still uses the html property when throwing an error is an ast thing, not a loop protector thing.  It might be a good idea for now to console.log or something before we delete the html property so that we can see what errors are using it.

**Comment by kevinbarabash** [comment]
@Potato42 I think you're right about the cause.  I going to create a separate bug for the issue with the blank "show me where" dialog.

**Comment by benjaminjkraft** [comment]
Also going to take a look at this, likewise to #522.

**Comment by benjaminjkraft** [comment]
Ok, I believe this is now fixed in d16b858; the infinite loop error now has the proper text.  Thanks for all the reports and repros.  I'm not 100% sure we fixed all the issues mentioned because a few of the repros were a bit flaky for me, so let me know if there were any parts of this that aren't fixed for you.
